### DISCUSSION TITLE
collectFile(): and collect(): a way to return an error message if collection is finally empty

### DISCUSSION THREAD

**Discussion by gael-millot** [nextflow-io/nextflow#3732] [Q&A] [answered]
Hi.

Thanks for your excellent work !
It would be important to have a new argument for collectFile(): and collect(), that returns an error message if collections by these functions are finally empty.
Or is there an alternative way to control this ?
Because for the moment, I do not know how to control this aspect when gathering outputs from parallele processes.
Thanks !

Gael.

**Comment by bentsherman** [comment, marked as answer]
You could use `count` and `subscribe` to check whether the incoming channel is empty before you call `collect` / `collectFile`. Here's an example:
```groovy
// items to collect
ch_samples = Channel.empty()

ch_samples
  | count
  | subscribe { n ->
    if ( n == 0 ) error "There are no items to collect"
  }

ch_samples
  | collect
```

**Comment by gael-millot** [comment]
Hi Ben.

Thanks a lot !
It works.

Best,

Gael.


De : Ben Sherman ***@***.***>
Envoyé : mercredi 8 mars 2023 16:04
À : nextflow-io/nextflow ***@***.***>
Cc : Gael MILLOT ***@***.***>; Author ***@***.***>
Objet : Re: [nextflow-io/nextflow] collectFile(): and collect(): a way to return an error message if collection is finally empty (Issue #3730)


You could use count and subscribe to check whether the incoming channel is empty before you call collect / collectFile. Here's an example:

// items to collect

ch_samples = Channel.empty()



ch_samples

  | count

  | subscribe { n ->

    if ( n == 0 ) error "There are no items to collect"

  }



ch_samples

  | collect
